docs/tokens.md

```diff
@@ -3,137 +3,239 @@ id: tokens
-- [`ERC721Mintable`](api/token/ERC721#erc721mintable) — like the ERC20 version, `ERC721Mintable` allows addresses with the `Minter` role to mint tokens.
-- [`ERC721Pausable`](api/token/ERC721#erc721pausable) — like the ERC20 version, `ERC721Pausable` allows addresses with the `Pauser` role to freeze transfers of tokens.
+The standard also bring multiple quality-of-life improvements, such as getting rid of the confusion around `decimals`, minting and burning with proper events, among others, but its killer feature are **send hooks**. A hook is simply a function in a contract that is called when tokens are sent, meaning **accounts and contracts can react to receiving tokens**.
 
+This enables a lot of interesting use cases, including atomic purchases using tokens (no need to do `approve` and `transferFrom` in two separate transactions), rejecting reception of tokens (by reverting on the hook call), and more generally brings tokens and Ether closer together by providing the equivalent of a `msg.value` field, but for tokens.
 
-We'll use these contracts to tokenize the time of our dogsitters: when a dogsitter wants to sell an hour of their time to watch a dog, they can mint an ERC721 token that represents that hour slot and then sell this token on an exchange. Then they'll go to the owner's house at the right time to watch their doggos.
+Furthermore, since contracts are required to implement these hooks in order to receive tokens, _no tokens can get stuck in a contract that is unaware of the ERC777 protocol_, as has happened countless times when using ERC20s.
 
-Here's what tokenized dogsitter timeframes might look like:
+### What if I already use ERC20?
+The standard has you covered! The ERC777 standard is **backwards compatible with ERC20**, meaning you can interact with these tokens as if they were ERC20, using the standard functions, while still getting all of the niceties, including send hooks. See the [EIP's Backwards Compatibility section](https://eips.ethereum.org/EIPS/eip-777#backward-compatibility) to learn more.
+
+### Constructing an ERC777 Token Contract
+
+We will replicate the `GLD` example of the [ERC20 guide](#constructing-an-erc20-token-contract), this time using ERC777. As always, check out the [`API reference`](api/token/ERC777) to learn more about the details of each function.
 
 ```solidity
-contract DoggoTime is ERC721Full {
-    using Counters for Counters.Counter;
-    Counters.Counter private tokenId;
+pragma solidity ^0.5.0;
 
-    constructor(
-        string memory name,
-        string memory symbol
-    )
-        ERC721Full(name, symbol)
-        public
-    {}
+import "openzeppelin-solidity/contracts/token/ERC777/ERC777.sol";
 
-    function createDoggoTimeframe(
-        string memory tokenURI
+contract GLDToken is ERC777 {
+    constructor(
+        uint256 initialSupply,
+        address[] memory defaultOperators
```

Comment:
What do you think about just using an empty array for `defaultOperators`? I think we can leave this for the advanced guide mentioned below.

Reply:
You mean passing an empty array to the constructor? I worry that may look a bit cryptic, at least the current version gives the concept a name that we can refer to.

```
ERC777("Gold", "GLD", [])
```

Talking about 'the empty array we passed' will make this harder to understand, imo.

Reply:
Hm, that's true.